### PR TITLE
Implement total in memory message count

### PR DESCRIPTION
**Pull request by a5anka**
Expose a metric to view the total enqueued message count.

Use metric manager to keep track of total enqueue count

Use metric manager to keep track of total enqueue count

N/A

- Messaging metric for total enqueued message count in a node

Tracked with #177

N/A

N/A

N/A

## Automation tests
 - Unit tests - 78%
 - Integration tests

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java [hunk 1]
```diff
@@ -64,7 +65,16 @@
     private BrokerHelper brokerHelper;
 
     public Broker(StartupContext startupContext) throws Exception {
-        this.messagingEngine = new MessagingEngine(startupContext.getService(DataSource.class));
+        MetricService metrics = startupContext.getService(MetricService.class);
+        if (Objects.nonNull(metrics)) {
```
Comment (abeykoon) [inline modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java]: What does this check do?

Comment (a5anka) [reply modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java]: We thought it is self-explanatory. :open_mouth: What do you think it does?

File: modules/broker-core/src/main/java/org/wso2/broker/core/QueueHandlerFactory.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+public class QueueHandlerFactory {
+    private final SharedMessageStore sharedMessageStore;
+    private final BrokerMetricManager metricManager;
+
+    public QueueHandlerFactory(SharedMessageStore sharedMessageStore, BrokerMetricManager metricManager) {
+        this.sharedMessageStore = sharedMessageStore;
+        this.metricManager = metricManager;
+    }
+
+    public QueueHandler createDurableQueue(String queueName, boolean autoDelete) throws BrokerException {
```
Comment (riyafa) [inline modules/broker-core/src/main/java/org/wso2/broker/core/QueueHandlerFactory.java]: Shall we add javadoc?

File: modules/broker-core/src/main/java/org/wso2/broker/core/QueueHandlerFactory.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        Queue queue = new DbBackedQueueImpl(queueName, autoDelete, sharedMessageStore);
+        return new QueueHandler(queue, metricManager);
+    }
+
+    public QueueHandler createNonDurableQueue(String queueName, int capacity, boolean autoDelete) {
```
Comment (riyafa) [inline modules/broker-core/src/main/java/org/wso2/broker/core/QueueHandlerFactory.java]: Shall we add javadoc?

File: modules/broker-core/src/main/java/org/wso2/broker/core/store/StoreFactory.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    private final DaoFactory daoFactory;
+    private final BrokerMetricManager metricManager;
+
+    public StoreFactory(DataSource dataSource, BrokerMetricManager metricManager) {
+        daoFactory = new DaoFactory(dataSource);
+        this.metricManager = metricManager;
+    }
+
+    public ExchangeRegistry getExchangeRegistry() {
```
Comment (riyafa) [inline modules/broker-core/src/main/java/org/wso2/broker/core/store/StoreFactory.java]: Shall we add javadoc to methods?

**Pull request merged**